src/libs/TransactionUtils.js

```diff
@@ -69,6 +69,7 @@ function buildOptimisticTransaction(amount, currency, reportID, comment = '', so
 function getUpdatedTransaction(transaction, transactionChanges, isFromExpenseReport) {
     // Only changing the first level fields so no need for deep clone now
     const updatedTransaction = _.clone(transaction);
+    let shouldStopSmartscan = false;
 
     // The comment property does not have its modifiedComment counterpart
```

Comment:
could we leave a comment explaining the smartscan is not stopped when Comment is changed?

@Gonals Also shouldnt the smartscan stop also when merchant is edited? this is not implemented in the App yet but its in the Auth so I can add it there

Reply:
Ah, it should, but we still didn't have merchants when I coded this. We do need to add it to auth (and eventually here)

Reply:
Actually, I just added it here too. No harm in doing so.
I've also created an issue to handle it in Auth too and will fix that tomorrow: https://github.com/Expensify/Expensify/issues/309894

Reply:
PR is up https://github.com/Expensify/Auth/pull/8549